plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerColumnHandle.java

```diff
@@ -117,40 +105,21 @@ private static boolean isCharacterColumn(ColumnMetadata column)
         return column.getType() instanceof CharType || column.getType() instanceof VarcharType || column.getType() instanceof VarbinaryType;
     }
 
-    private static Object propertyValue(ColumnMetadata column, String property)
-    {
-        try {
-            return Literal.parse((String) column.getProperties().get(property), column.getType());
-        }
-        catch (IllegalArgumentException e) {
-            throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `%s` property must be a valid %s literal".formatted(property, column.getType().getDisplayName()), e);
-        }
-    }
-
     private static ValueSet stepValue(ColumnMetadata column)
     {
         Type type = column.getType();
-        String rawStep = (String) column.getProperties().get(STEP_PROPERTY);
-        if (rawStep == null) {
+        Object step = PropertyValues.propertyValue(column, STEP_PROPERTY);
+        if (step == null) {
             return ValueSet.none(type);
         }
         if (isCharacterColumn(column)) {
             throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `%s` property cannot be set for CHAR, VARCHAR or VARBINARY columns".formatted(STEP_PROPERTY));
         }
-        if (DATE.equals(column.getType()) || type instanceof TimestampType || type instanceof TimestampWithTimeZoneType || type instanceof TimeType || type instanceof TimeWithTimeZoneType) {
-            try {
-                return ValueSet.of(BIGINT, Duration.valueOf(rawStep).roundTo(TimeUnit.NANOSECONDS));
-            }
-            catch (IllegalArgumentException e) {
-                throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `%s` property for a %s column must be a valid duration literal".formatted(STEP_PROPERTY, column.getType().getDisplayName()), e);
-            }
-        }
-        try {
-            return ValueSet.of(type, Literal.parse(rawStep, type));
-        }
-        catch (IllegalArgumentException e) {
-            throw new TrinoException(INVALID_COLUMN_PROPERTY, "The `%s` property for a %s column must be a valid %s literal".formatted(STEP_PROPERTY, column.getType().getDisplayName(), type.getDisplayName()), e);
+        Type stepType = type;
+        if (DATE.equals(type) || type instanceof TimestampType || type instanceof TimestampWithTimeZoneType || type instanceof TimeType || type instanceof TimeWithTimeZoneType) {
+            stepType = BIGINT;
```

Comment:
I think we only support long steps, maybe no need to check the types here

Reply:
it was failing without that